Plugins/Events/Events/ItemEvents.cpp

```diff
@@ -149,4 +154,80 @@ void ItemEvents::RemoveItemHook(Services::Hooks::CallType type, CItemRepository*
     Events::SignalEvent(before ? "NWNX_ON_ITEM_INVENTORY_REMOVE_ITEM_BEFORE" : "NWNX_ON_ITEM_INVENTORY_REMOVE_ITEM_AFTER", thisPtr->m_oidParent);
 }
 
+uint32_t ItemEvents::FindItemWithBaseItemIdHook(CItemRepository* thisPtr, uint32_t baseItem, int32_t nTh)
+{
+    auto *pItemHolder = Utils::AsNWSCreature(Globals::AppManager()->m_pServerExoApp->GetGameObject(thisPtr->m_oidParent));
+
+    if(!pItemHolder)
+    {
+        // For our purposes we only want this to be used on creature ItemRepositories
+        return m_FindItemWithBaseItemIdHook->CallOriginal<int32_t>(thisPtr, baseItem, nTh);
+    }
+
+    // We're most assuredly in an infinite loop here caused by the scripting end
+    // unless the player has > 255 stacks of ammo they can't equip
+    // Suppose I could set this to a higher value but I think this is safe
+    if(nTh > 255)
+    {
+        return OBJECT_INVALID;
+    }
+
+    auto ItemSanityCheck = [&](uint32_t objectId) -> uint32_t {
```

Comment:
A "sanity check" function should probably just return a bool, and then caller can CallOriginal() if the check failed.